pkg/operator/ceph/csi/template/rbd/csi-rbdplugin-provisioner-dep.yaml

```diff
@@ -268,10 +298,56 @@ spec:
               mountPath: /csi
           imagePullPolicy: {{ .ImagePullPolicy }}
         {{ end }}
+        {{ if .CSILogRotation }}
+        - args:
+          - |
+            echo "Starting the csi-logrotate-sidecar"
+            mkdir -p /logrotate-config;
+            echo '/var/lib/cephcsi/log/*.log {
+                daily
+                missingok
+                rotate 7
+                compress
+                copytruncate
+                notifempty
+            }' > /logrotate-config/csi;
+            echo "File creation container completed";
+            LOG_ROTATE_CEPH_FILE=/logrotate-config/csi
+            PERIODICITY={{ .CSILogRotationPeriod }}
+            if [ "$PERIODICITY" != "" ]; then
+              echo "replace default daily with given user input"
+              sed --in-place "s/daily/$PERIODICITY/g" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            LOG_MAX_SIZE={{ .CSILogRotationMaxSize }}
+            if [ "$LOG_MAX_SIZE" != "0" ]; then
+              sed --in-place "4i \ \ \ \ maxsize $LOG_MAX_SIZE" "$LOG_ROTATE_CEPH_FILE"
+            fi
+            while true; do
+              logrotate --verbose "$LOG_ROTATE_CEPH_FILE"
+              sleep 15m
+            done
+          command:
+          - /bin/sh
+          - -c
+          image: {{ .CSIPluginImage }}
+          imagePullPolicy: IfNotPresent
+          name: log-collector
+          volumeMounts:
+          - mountPath:  /logrotate-config
+            name: csi-logs-logrotate
+          - mountPath: /var/lib/cephcsi/log/
+            name: csi-log
+        {{ end }}
       volumes:
         - name: host-dev
           hostPath:
             path: /dev
+        - name: csi-log
+          hostPath:
+            path: /var/lib/cephcsi/log/
+        - name: csi-logs-logrotate
+          hostPath:
+            path: /etc/logrotate.d
```

Comment:
why we are mounting /etc/logrotation of hostPath here?

Reply:
The sidecar is created using 

```
- mountPath: /logrotate-config
    name: csi-logs-logrotate
```

So we need a volume to have a hostPath.

Reply:
but we do we need to have a HostPath for this one, can you be more specific about it, we can't write anything to `/etc/logrotate.d` folder as we are not using logrotate from the node system.

Reply:
I thought we should create the logrotate config file in the same folder